cmd/otel-allocator/README.md

```diff
@@ -4,6 +4,136 @@ The TargetAllocator is an optional separately deployed component of an OpenTelem
 distribute targets of the PrometheusReceiver on all deployed Collector instances. The release version matches the
 operator's most recent release as well.
 
+In essence, Prometheus Receiver configs are "stolen" and replaced with a http_sd_configs directive that points to the 
+Allocator, these are then loadbalanced/sharded to the collectors. In addition, the TargetAllocator can discover targets
+via Prometheus CRs (currently ServiceMonitor, PodMonitor) which it presents as scrape configs and jobs on the
+`/scrape_configs` and `/jobs` endpoints respectively.
+
+# Usage
+The `targetAllocator:` controls the TargetAllocator general properties. Full API spec can be found here: [api.md#opentelemetrycollectorspectargetallocator](../../docs/api.md#opentelemetrycollectorspectargetallocator)
+
+A basic example that deploys, but does nothing.
+```yaml
+apiVersion: opentelemetry.io/v1alpha1
+kind: OpenTelemetryCollector
+metadata:
+  name: collector-with-ta
+spec:
+  mode: statefulset
+  targetAllocator:
+    enabled: true
+  config: |
+    receivers:
+      prometheus:
+        config:
+          scrape_configs:
+          - job_name: 'otel-collector'
+
+    exporters:
+      logging:
+
+    service:
+      pipelines:
+        traces:
+          receivers: [prometheus]
+          processors: []
+          exporters: [logging]
+```
+
+## PrometheusCR specifics
+TargetAllocator discovery of PrometheusCRs can be turned on with
+```yaml
+  targetAllocator:
+    enabled: true
+    prometheusCR:
+      enabled: true
+```
+The CRs can be filtered by labels as documented here: [api.md#opentelemetrycollectorspectargetallocatorprometheuscr](../../docs/api.md#opentelemetrycollectorspectargetallocatorprometheuscr)
+
+The prometheus receiver in the deployed collector also has to know where the Allocator service exists. This is done by a
+OpenTelemetry Collector operator specific config.
+```yaml
+  config: |
+    receivers:
+      prometheus:
+        config:
+          scrape_configs:
+          - job_name: 'otel-collector'
+        target_allocator:
+          endpoint: http://my-targetallocator-service
+          interval: 30s
+          collector_id: "${POD_NAME}"
+```
+Upstream documentation here: [Prometheusreceiver](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/receiver/prometheusreceiver#opentelemetry-operator)
+
+The TargetAllocator service is named based on the OpenTelemetryCollector CR name. `collector_id` should be unique per
+collector instance, such as the pod name. The `POD_NAME` environment variable is convenient since this is supplied
+to collector instance pods by default.
+
+The CRDs naturally also has to exist for the Allocator to pick them up. The best place to get them is from
```

Comment:
```suggestion
The Prometheus CRDs also have to exist for the Allocator to pick them up. The best place to get them is from
```

Otherwise I'd agree with @jaronoff97 to perhaps leave this out for now and do it in follow up?

Reply:
I think it should be mentioned that this is not done by automagic (which is very common in the k8s world). It can be removed later in favour of a link to a proper doc. Accepting your change.